src/shell/gmenu.js

```diff
@@ -92,12 +93,19 @@ export class ListBox extends PopupMenu.PopupMenuSection {
         this.actor.add_child(this.box);
 
         // Submenu Container
-        this.sub = new St.BoxLayout({
-            clip_to_allocation: true,
-            vertical: false,
-            visible: false,
-            x_expand: true,
-        });
+        this.sub = Number(PACKAGE_VERSION.split('.')[0]) >= 48
+            ? new St.BoxLayout({
+                clip_to_allocation: true,
+                orientation: Clutter.Orientation.HORIZONTAL, // GNOME 48
+                visible: false,
+                x_expand: true,
+            })
+            : new St.BoxLayout({
+                clip_to_allocation: true,
+                vertical: false, // GNOME 46/47 (afaict, these don't have orientation)
```

Comment:
I believe they do, actually. `Clutter.Orientation` is definitely defined in GNOME 47 (just tested in `gjs`), and `St.BoxLayout.vertical` has been [documented](https://gnome.pages.gitlab.gnome.org/gnome-shell/st/property.BoxLayout.vertical.html) as "A convenience property for the `ClutterBoxLayout:vertical` property of the internal layout for `StBoxLayout`." for as long as it's existed. (Those docs were generated in February 2024, in fact. Which is why they still don't show `St.BoxLayout.vertical` as deprecated.)

Reply:
Er, half-finished thought. I meant to say, it's been documented [alongside the `orientation` property](https://gnome.pages.gitlab.gnome.org/gnome-shell/st/property.BoxLayout.orientation.html) for as long as it's existed.

Reply:
Nope, I'm wrong. `orientation` [was only added a month ago](https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/01541929eb02557f213d22d481388896bf6fc746).

The docs were generated with the 2024-02 version _of gi-docgen_, they're not themselves a year old. Whoops.